lib/lm_docstache/parser.rb

```diff
@@ -140,6 +157,32 @@ def replace_variables_in_document!
       end
     end
 
+    def replace_data_sequentially_in_document!
+      data_sequential_replacement.each do |tag_key, values|
+
+        tag = Regexp.escape("{{#{tag_key}}}")
+        pattern_found = 0
+
+        document.css('w|t').each do |text_node|
+          text = text_node.text
+
+          if text.match(tag)
+
+            text.gsub!(/#{tag}/) do |_match|
+              value = values[pattern_found]
+              # if there is no more available value replace the content with empty string
+              return '' unless value
+
+              pattern_found +=1
+              value
+            end
+
+            text_node.content = text
+          end
+        end
+      end
+    end
```

Comment:
As a suggestion to make things simpler, I suggest the following:
(for instance, we actually don't need the counter)
```suggestion
    def replace_data_sequentially_in_document!
      data_sequential_replacement.each do |tag_key, values|
        values = values.dup
        tag = Regexp.escape("{{#{tag_key}}}")

        document.css('w|t').each do |text_node|
          next unless (text = text_node).text.match(tag)

          text.gsub!(/#{tag}/) { |_match| values.any? ? values.shift : '' }
          text_node.content = text
        end
      end
    end
```